src/adapter/src/optimize/materialized_view.rs

```diff
@@ -266,6 +274,9 @@ impl Optimize<GlobalMirPlan> for Optimizer {
         )
         .map_err(OptimizerError::Internal)?;
 
+        // Trace the pipeline output under `optimize`.
+        trace_plan(&df_desc);
```

Comment:
Recording the `optimize/global` and final `optimize` entries in the `OptimizerTrace` now happens inside the `Optimize` implementations and no longer in the `Coordinator` methods.